### PR TITLE
Add simple nested editor to Tabulator

### PR DESCRIPTION
**Pull request by hoxbro**
https://github.com/user-attachments/assets/66e34b30-5bc4-4ed2-b6f7-ba1865ce5c22

``` python
import pandas as pd
import panel as pn

pn.extension("tabulator")

df = pd.DataFrame({"0": ["A", "B", "A"], "1": [1, 2, 3], "2": [None, None, None]})

options = {
    "A": [1, 2, 3, 4, 5],
    "B": {
        "1": list(range(5, 10)),
        "2": list(range(10, 15)),
        "3": list(range(15, 20)),
    },
}
tabulator_editors = {
    "0": {"type": "list", "values": ["A", "B"]},
    "1": {"type": "list", "values": [1, 2, 3]},
    "2": {"type": "nested", "options": options, "lookup_order": ["0", "1"]},
}

def on_edit(event):
    if event.column in ("0", "1"):
        edit_table.patch({"2": [(event.row, None)]})

edit_table = pn.widgets.Tabulator(df, editors=tabulator_editors, show_index=False, on_edit=on_edit)

edit_table
```

### CODE REVIEW

File: panel/models/tabulator.ts [hunk 1]
```diff
@@ -302,6 +302,20 @@ const datetimeEditor = function(cell: any, onRendered: any, success: any, cancel
   return input
 }
 
+const nestedEditor = function(cell: any, editorParams: any) {
+  //cell - the cell component for the editable cell
+
+  const row = cell.getRow().getData()
+  let values = editorParams.options
+  for (const i of editorParams.lookup_order) {
+    values = values[row[i]]
```
Comment (philippjfr) [inline panel/models/tabulator.ts:311]: What happens if there is no entry here? Does it handle it gracefully?

**Review by philippjfr — APPROVED** [review]
Asked on clarifying question, otherwise this looks great!

**Comment by MarcSkovMadsen** [comment]
This is an alternative example in an attempt to make it more clear which columns contains the nested selections and how the options depends on the selections in other columns.

```python

import pandas as pd
import panel as pn

pn.extension("tabulator")

df = pd.DataFrame({"0": ["A", "B", "A"], "1": [1, 2, 3], "Nested Selection": [None, None, None]})

options = {
    "A": ["A.1", "A.2", "A.3", "A.4", "A.5"],
    "B": {
        "1": ["B1.1", "B1.2", "B1.3"],
        "2": ["B2.1", "B2.2", "B2.3"],
        "3": ["B3.1", "B3.2", "B3.3"],
    },
}
tabulator_editors = {
    "0": {"type": "list", "values": ["A", "B"]},
    "1": {"type": "list", "values": [1, 2, 3]},
    "Nested Selection": {"type": "nested", "options": options, "lookup_order": ["0", "1"]},
}


def on_edit(event):
    if event.column in ("0", "1"):
        edit_table.patch({"Nested Selection": [(event.row, None)]})


edit_table = pn.widgets.Tabulator(df, editors=tabulator_editors, show_index=False, on_edit=on_edit)

edit_table.servable()
```

**Comment by hoxbro** [comment]
I think it is a better example, so you are welcome to open a PR with your suggested changes.

**Pull request merged**